scripts/core/editor3/components/MultipleHighlights.jsx

```diff
@@ -119,70 +119,132 @@ export class MultipleHighlights extends React.Component {
         return state;
     }
 
-    addHighlight(highlightType, highlightData) {
-        // NICE-TO-HAVE: remove associated data when highlights are removed by deleting characters in the editor
+    setHighlightsState(editorState, highlightsState) {
+        // TODO: save to first block
+        this.setState(highlightsState);
```

Comment:
If it's stateless we can remove all instances of `this.state` or `this.setState`. So this function would simply return a new `editorState`

Reply:
Yes, this is the intention, because of this I added the TODO. For now I just added some code in order to keep the comments working. We can implement the set and get here or we can receive the functions in props.

Reply:
I think the component is not that big and it would be better to perform the refactoring in one go, than to merge this half-baked PR. I'm not sure if that's what you intend to do.

Reply:
Yes, the intention was to keep the code working, and have something to review and agree on.
And yes, we can do this in a single PR.